addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -1,484 +1,711 @@
+            if (!formInfo || !formInfo.fields) {
+                return;
+            }
+            const proms = formInfo.fields.map(field => this._getFieldRecords(field));
+            return Promise.all(proms).then(() => {
+                formInfo.fields.forEach(field => {
+                    let option;
+                    switch (field.type) {
+                        case 'many2one':
+                            option = this._buildSelect(field);
+                            break;
+                        case 'char':
+                            option = this._buildInput(field);
+                            break;
                     }
-                });
-
-                var save = function () {
-                    var selectedFieldName = this.$el.find("[name='field_selection']").val();
-                    var selectedField = fields[selectedFieldName];
-                    self.append_field(selectedField);
-                };
-
-                var dialog = new FormEditorDialog(self, {
-                    title: 'Field Parameters',
-                    size: 'medium',
-                    $content: '<form role="form">' + fieldSelection + '</form>',
-                }).open();
-                dialog.on('save', this, ev => {
-                    ev.stopPropagation();
-                    save.call(dialog);
+                    uiFragment.insertBefore(option, firstOption);
                 });
             });
         },
-
-        // Create a custom field
-        website_form_custom_field: function (previewMode, widgetValue, params) {
-            var default_field_name = 'Custom ' + this.el.querySelector(`[data-website_form_custom_field="${widgetValue}"]`).textContent; // TODO improve
-            this.append_field({
-                name: default_field_name,
-                string: default_field_name,
-                custom: true,
-                type: widgetValue,
-                // Default values for x2many fields
-                records: [
-                    {
-                        id: 'Option 1',
-                        display_name: 'Option 1'
-                    },
-                    {
-                        id: 'Option 2',
-                        display_name: 'Option 2'
+        /**
+         * Returns a we-select element with field's records as it's options
+         *
+         * @private
+         * @param {Object} field
+         * @return {HTMLElement}
+         */
+        _buildSelect: function (field) {
+            const selectEl = options.buildElement('we-select', null, {
+                dataAttributes: {
+                    noPreview: 'true',
+                    fieldName: field.name,
+                    isSelect: 'true',
+                },
+            });
+            const noneButton = options.buildElement('we-button', 'None', {
+                dataAttributes: {
+                    addActionField: 0,
+                },
+            });
```

Comment:
Mmh, I don't know why I exposed buildElement. Best to probably not use it and simply build the element using standard javascript by yourself, my goal was not to reimplement jQuery by myself ^^